src/dbnode/network/server/tchannelthrift/cluster/service.go

```diff
@@ -140,6 +140,8 @@ func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryR
 			return nil, convert.ToRPCError(err)
 		}
 
+		defer results.Finalize()
```

Comment:
we were also leaking resources on errors :(

Reply:
Ah good catch.